src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateDataRetentionService.kt

```diff
@@ -62,6 +64,8 @@ class CertificateDataRetentionService(
         logger.info { "Finding certificates with sourceType $sourceType to remove final retention period data from" }
         with(certificateRepository.findPendingRemovalOfFinalRetentionData(sourceType = sourceType)) {
             forEach {
+                // TODO - EIP1-4481 - get s3 arn from certificate instead of print request
+                s3CertificatePhotoService.removePhoto(it.printRequests[0].photoLocationArn!!)
                 certificateRepository.delete(it)
```

Comment:
I'd like to discuss the efficiency of iterating over each certificate and deleting from both S3 and the database each time.

Reply:
The way we coped with the transaction boundary within VCA was to use ApplicationRemovalTransactionalService. I think a similar approach here would work to prevent rolling back the whole list. 
Also, in this scenario because we're not using SQS to independently process each certificate, we might need a try/catch(Exception) inside the for each block so that if we do get blocked deleting one certificate we can continue to process the rest.